runners/prism/java/src/main/java/org/apache/beam/runners/prism/PrismRunner.java

```diff
@@ -72,15 +82,44 @@ public PipelineResult run(Pipeline pipeline) {
         prismPipelineOptions.getDefaultEnvironmentType(),
         prismPipelineOptions.getJobEndpoint());
 
-    return internal.run(pipeline);
+    try {
+      PrismExecutor executor = startPrism();
+      PortableRunner delegate = PortableRunner.fromOptions(prismPipelineOptions);
+      return new PrismPipelineResult(delegate.run(pipeline), executor::stop);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  PrismExecutor startPrism() throws IOException {
+    PrismLocator locator = new PrismLocator(prismPipelineOptions);
+    int port = findAvailablePort();
+    String portFlag = String.format(PrismExecutor.JOB_PORT_FLAG_TEMPLATE, port);
+    String serveHttpFlag =
+        String.format(
+            PrismExecutor.SERVE_HTTP_FLAG_TEMPLATE, prismPipelineOptions.getEnableWebUI());
```

Comment:
Just need to add the --
idle_shutdown_timeout=5m flag

(That is, shutdown the prism instance if it's been idle for 5 minutes) to avoid dangling processes. We can make this shorter too TBH, but lets start conservative.